redaxo/src/addons/structure/lib/select_category.php

```diff
@@ -133,9 +132,9 @@ protected function outGroup($parentId, $level = 0)
             $name = $option[0];
             $value = $option[1];
             $id = $option[2];
-            if (0 == $id || !$this->checkPerms || ($this->checkPerms && rex::getUser()->getComplexPerm('structure')->hasCategoryPerm($option[2]))) {
+            if (0 == $id || !$this->checkPerms || (rex::getUser()->getComplexPerm('structure')->hasCategoryPerm($option[2]))) {
                 $ausgabe .= $this->outOption($name, $value, $level);
-            } elseif (($this->checkPerms && rex::getUser()->getComplexPerm('structure')->hasCategoryPerm($option[2]))) {
+            } elseif ($this->checkPerms && rex::getUser()->getComplexPerm('structure')->hasCategoryPerm($option[2])) {
```

Comment:
lt. rexstan ist das ganze `elseif` redundant.. ich würde es erstmal so lassen ;)

Reply:
Sagt rexstan nicht nur, dass der Teil nach `&&` überflüssig ist?
Ich denke, da stimme ich auch zu. Also mir scheint, `elseif ($this->checkPerms) {` würde genügen.
Aber für mich auch ok, es lieber so zu lassen.

Reply:
hmm je mehr ich drüber nachdenke, je mehr galube ich dass das ELSE-IF dead ist

Reply:
<img width="382" alt="grafik" src="https://user-images.githubusercontent.com/120441/194704421-15e8cd9d-9186-4440-8bed-bb205c6740a3.png">

Reply:
Hmm stimmt. Dann wäre jetzt höchstens die Frage, was wollte man damit eigentlich erreichen. Also ob es wirklich weg kann, oder eigentlich die Bedingung eine andere sein sollte. Das fällt mir gerade schwer einzuschätzen.

Andererseits könnte man auch sagen, dass es bisher ja nicht negativ aufgefallen ist, dass der Code nie ausgeführt wird.

Oder vielleicht auskommentieren mit Kommentar dran?

Reply:
ich lass es erstmal so und wenn mich die lust treibt, würde man ggf. erstmal nen unit test machen :)